src/tempo.py

```diff
@@ -201,39 +141,22 @@ def _build_storage_config(self, s3_config: dict):
                 queue_depth=20000,
             ),
             backend="s3",
-            s3=tempo_config.S3(
-                bucket=s3_config["bucket"],
-                access_key=s3_config["access-key"],
-                endpoint=s3_config["endpoint"],
-                secret_key=s3_config["secret-key"],
-            ),
+            s3=tempo_config.S3(**s3_config),
             # starting from Tempo 2.4, we need to use at least parquet v3 to have search capabilities (Grafana support)
             # https://grafana.com/docs/tempo/latest/release-notes/v2-4/#vparquet3-is-now-the-default-block-format
             block=tempo_config.Block(version="vParquet3"),
         )
         return tempo_config.Storage(trace=storage_config)
 
-    def is_ready(self):
-        """Whether the tempo built-in readiness check reports 'ready'."""
-        if self.use_tls:
-            tls, s = f" --cacert {self.server_cert_path}", "s"
-        else:
-            tls = s = ""
-
-        # cert is for fqdn/ingress, not for IP
-        cmd = f"curl{tls} http{s}://{self._external_hostname}:{self.tempo_http_server_port}/ready"
-
-        try:
-            out = getoutput(cmd).split("\n")[-1]
-        except (CalledProcessError, IndexError):
-            return False
-        return out == "ready"
-
     def _build_querier_config(self, roles_addresses: Dict[str, Set[str]]):
         """Build querier config"""
-        addr = "localhost"
-        if TempoRole.query_frontend in roles_addresses.keys():
-            addr = roles_addresses[TempoRole.query_frontend].pop()
+        # if distributor and query-frontend have the same address, then the mode of operation is 'all'.
```

Comment:
Well, in that case it will fetch the `query_frontend` address from the address set of `query_frontend` and not use `localhost`. Since the set of addrs for distributors will be different than the set of addrs for query frontends. 
This can only be one given address not a list, as Tempo supports.